R/function_left_parentheses_linter.R

```diff
@@ -60,7 +60,15 @@ function_left_parentheses_linter <- function() { # nolint: object_length.
       source_expression = source_expression,
       lint_message = "Remove spaces before the left parenthesis in a function call.",
       range_start_xpath = "number(./@col2 + 1)", # start after function / fun
-      range_end_xpath = "number(./following-sibling::OP-LEFT-PAREN/@col1 - 1)" # end before (
+      # if '(' is on the same line, end before '(', otherwise, just set range[2]=range[1].
+      # this is a janky XPath 1.0 implementation for if(same line, "before (", "range[1]")
+      #   by converting if(COND, x, y) to 1[COND] * x + (1 - 1[COND]) * y for 1[.] an indicator function
+      range_end_xpath = "
```

Comment:
alternatively, we might change the lint message to something like "Move the left parenthesis to the same line as the start of the function call."

Reply:
Are we vectorized on `range_end_xpath`?
This sounds like we should do a branch for each of the two lints (wrong col on same line or wrong line).

Reply:
We should be, but it makes sense to add a test as such.

> branch for each of the two lints

yep, that's what i was getting at w/ changing the lint message.

Reply:
Went that route now